Subject: rt: xfs fix
From: Ingo Molnar <mingo@elte.hu>
Date: Sun Feb 08 17:26:51 CET 2009

XFS still relies on semaphore implementation internals:

 fs/xfs/linux-2.6/xfs_buf.c: In function ‘xfs_buf_lock_value’:
 fs/xfs/linux-2.6/xfs_buf.c:846: error: incompatible types in return

Convert it to a compat semaphore. Loses hard-real-time priority inheritance
support this way.

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 fs/xfs/linux-2.6/xfs_buf.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6-tip/fs/xfs/linux-2.6/xfs_buf.h
===================================================================
--- linux-2.6-tip.orig/fs/xfs/linux-2.6/xfs_buf.h
+++ linux-2.6-tip/fs/xfs/linux-2.6/xfs_buf.h
@@ -145,7 +145,7 @@ typedef int (*xfs_buf_bdstrat_t)(struct 
 #define XB_PAGES	2
 
 typedef struct xfs_buf {
-	struct semaphore	b_sema;		/* semaphore for lockables */
+	struct compat_semaphore	b_sema;		/* semaphore for lockables */
 	unsigned long		b_queuetime;	/* time buffer was queued */
 	atomic_t		b_pin_count;	/* pin count */
 	wait_queue_head_t	b_waiters;	/* unpin waiters */
